internal-client/src/main/java/org/apache/uniffle/client/impl/grpc/GrpcClient.java

```diff
@@ -76,9 +77,13 @@ protected GrpcClient(ManagedChannel channel) {
     this.channel = channel;
   }
 
-  protected ByteBufAllocator createByteBufAllocator(
+  private static synchronized ByteBufAllocator getByteBufAllocator(
```

Comment:
To unify the structure and naming in this project as a whole, add a method called `getSharedPooledByteBufAllocator` into `GrpcNettyUtils` refer to https://github.com/apache/incubator-uniffle/pull/1803.

I think it's better to put the logic there.

Reply:
I rename to getSharedPooledByteBufAllocator, and move to GrpcNettyUtils.
I think for a process have only one config, so set only one cache instance.
And another issue is that do we need `ss.client.netty.client.shared.allocator.enabled` to enable cache allocator? For me, I think only one allocator is enough for grpc client, so I did not use `ss.client.netty.client.shared.allocator.enabled`

Reply:
I prefer to use `rss.client.netty.client.shared.allocator.enabled` to control whether to use a shared Netty allocator for gRPC as well. It's like what Spark does. Spark set `shared.allocator.enabled` to `true` **_always_**. In this case, we can unify the code in the project.

But I'm also okay with not using `rss.client.netty.client.shared.allocator.enabled` for gRPC. Because gRPC will be deprecated in the future in Uniffle, so I'm open to whatever happens to gRPC related codes.

Reply:
I'm fine with this. Yeah, it will always be `true` for Netty mode in Uniffle and Spark.